### PR TITLE
Adding support for Github API key #157

### PR DESCRIPTION
**Pull request by dhruvinsh**
resolves MordechaiHadad/bob#157

### CODE REVIEW

**Comment by dhruvinsh** [comment]
@MordechaiHadad First time trying rust lang. Please take a look at implementation and it works as expected for me.

Usage (one of the example):
```bash
GITHUB_TOKEN=<your token> bob update -a
```
or as normal usage,
```bash
bob update -a
```

File: src/cli.rs [hunk 1]
```diff
@@ -8,9 +8,31 @@ use crate::{
 use anyhow::Result;
 use clap::{Args, CommandFactory, Parser};
 use clap_complete::Shell;
-use reqwest::Client;
+use reqwest::{Client, Error};
 use tracing::{error, info};
 
+fn create_reqwest_client() -> Result<Client, Error> {
+    // fetch env variable
+    let github_token = match std::env::var("GITHUB_TOKEN") {
```
Comment (MordechaiHadad) [inline src/cli.rs:16]: Instead of just doing so keep it as Result<String> and then
```rs
if let Ok(v) = github_token {
// le code 
}
```

**Review by MordechaiHadad — CHANGES_REQUESTED** [review]

File: src/cli.rs (hunk 1 above)
Comment (dhruvinsh) [reply src/cli.rs:16]: As far as I understand rust, the purpose of current implementation is to have environment value or to have an empty string if not present.

And based on that make further logic for header values.

If it's not valid can you please explain in a bit of detail.

Comment (dhruvinsh) [reply src/cli.rs:16]: Never mind I think I got you. Let me give that a try.

Comment (dhruvinsh) [reply src/cli.rs]: Pushed new changes, please review.

**Review by MordechaiHadad — APPROVED** [review]

**Comment by MordechaiHadad** [comment]
> > Also if you can, add to the readme that it is possible to add ur github token to be used with bob
> 
> done.

Move it to configuration part in the README, and test if it works correctly if the env var doesn't exist.
after that it is ready for merge!

**Comment by MordechaiHadad** [comment]
Beautiful thank you @dhruvinsh

**Pull request merged**